docs/streams/core-concepts.html

```diff
@@ -172,6 +172,37 @@ <h2><a id="streams_processing_guarantee" href="#streams_processing_guarantee">Pr
         More details can be found in the <a href="/{{version}}/documentation#streamsconfigs"><b>Kafka Streams Configs</b></a> section.
     </p>
 
+    <h3><a id="streams_out_of_ordering" href="#streams_out_of_ordering">Out-of-Order Handling</a></h3>
+
+    <p>
+        Besides the guarantee that each record will be processed exactly-once, another issue that many stream processing application will face is how to
+        handle <a href="tbd">out-of-order data</a> that may impact their business logic. In Kafka Streams, there are two causes that could potentially
+        result in out-of-order data arrivals with respect to their timestamps:
+    </p>
+
+    <ul>
+        <li> Within a topic-partition, a record's timestamp may not be monotonically increasing along with their offsets. Since Kafka Streams will always try to process records within a topic-partition to follow the offset order, it can cause records with larger timestamps
```

Comment:
perhaps clarify to (and fix typo): "it can cause records with larger timestamps (but smaller offsets) to be processed earlier than records with smaller timestamp[s] (but larger offsets) in the same topic-partition"?